src/main/java/nextstep/auth/filters/TokenAuthenticationFilter.java

```diff
@@ -0,0 +1,45 @@
+package nextstep.auth.filters;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import nextstep.auth.authentication.AuthenticationToken;
+import nextstep.auth.filters.provider.AuthenticationProvider;
+import nextstep.auth.token.JwtTokenProvider;
+import nextstep.auth.token.TokenRequest;
+import nextstep.auth.token.TokenResponse;
+import nextstep.auth.user.UserDetails;
+import org.springframework.http.MediaType;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.stream.Collectors;
+
+public class TokenAuthenticationFilter extends AuthenticationRespondingFilter {
+    private final JwtTokenProvider jwtTokenProvider;
+
+    public TokenAuthenticationFilter(AuthenticationProvider<AuthenticationToken> authenticationProvider, JwtTokenProvider jwtTokenProvider) {
+        super(authenticationProvider);
+        this.jwtTokenProvider = jwtTokenProvider;
+    }
+
+    @Override
+    protected AuthenticationToken convert(HttpServletRequest request) throws IOException {
+        String content = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+        TokenRequest tokenRequest = new ObjectMapper().readValue(content, TokenRequest.class);
+
+        String principal = tokenRequest.getEmail();
```

Comment:
인터페이스(혹은 믹스인 인터페이스)를 활용하면 특정 DTO에 의존하지 않고 핵심 업무 규칙을 가지는 인터페이스를 이용해서 의존성의 흐름을 제어할 수 있을 것 같네요.
```suggestion
        String principal = authenticationSupplier.principal();
        String credential= authenticationSupplier.credential();
```
이런식으로 말이죠 ㅎㅎ